File: Examples/objective-c/TCMaskPopView/TCMaskPopView/ViewController.m
//
//  ViewController.m
//  TCMaskPopView
//
//  Created by Xin Zeng on 2/7/17.
//  Copyright © 2017 TinyCrayon. All rights reserved.
//

#import "ViewController.h"
@import TCMask;

@interface ViewController ()
@property (weak, nonatomic) IBOutlet UIImageView *imageView;

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view, typically from a nib.
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

- (void)viewDidAppear:(BOOL)animated {
    [super viewDidAppear:animated];
    if (_imageView.image == nil) {
        [self presentTCMaskView];
    }
}

- (IBAction)editButtonTapped:(id)sender {
    [self presentTCMaskView];
}

- (void)presentTCMaskView {
    UIImage *image = [UIImage imageNamed:@"Balloon.JPEG"];
    TCMaskView *maskView = [[TCMaskView alloc] initWithImage:image];
    maskView.delegate = self;
    [maskView presentFromRootViewController:self animated:true];
}

- (void)tcMaskViewDidCompleteWithMask:(TCMask *)mask image:(UIImage *)image {
    _imageView.image = [mask cutoutWithImage:image resize:true];
}

@end

File: Examples/swift/TCMaskPopView/TCMaskPopView/ViewController.swift
//
//  The MIT License
//
//  Copyright (C) 2016 TinyCrayon.
//
//  Permission is hereby granted, free of charge, to any person obtaining a copy
//  of this software and associated documentation files (the "Software"), to
//  deal in the Software without restriction, including without limitation the
//  rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
//  sell copies of the Software, and to permit persons to whom the Software is
//  furnished to do so, subject to the following conditions:
//
//  The above copyright notice and this permission notice shall be included in
//  all copies or substantial portions of the Software.
//
//  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
//  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
//  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
//  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
//  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
//  FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
//  IN THE SOFTWARE.
//

import UIKit
import TCMask

class ViewController: UIViewController, TCMaskViewDelegate {

    @IBOutlet weak var imageView: UIImageView!
    
    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view, typically from a nib.
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }
    
    override func viewDidAppear(_ animated: Bool) {
        if (imageView.image == nil) {
            presentTCMaskView()
        }
    }
    
    @IBAction func editButtonTapped(_ sender: Any) {
        presentTCMaskView()
    }

    func presentTCMaskView() {
        let image = UIImage(named: "Balloon.JPEG")!
        let maskView = TCMaskView(image: image)
        maskView.delegate = self
        maskView.presentFrom(rootViewController: self, animated: true)
    }
    
    func tcMaskViewDidComplete(mask: TCMask, image: UIImage) {
        imageView.image = mask.cutout(image: image, resize: true)
    }
}

